web/src/frontend/src/nav-soknad/components/vedlegg/Vedlegg.tsx

```diff
@@ -28,62 +32,100 @@ class Vedlegg extends React.Component<Props, State> {
 	constructor(props: Props) {
 		super(props);
 		this.state = {
-			vedlegg: []
+			vedleggId: null
 		};
 	}
 
+	hentVedleggsId(faktumKey: string): number {
+		const { fakta, vedlegg } = this.props;
+		const faktum = finnFaktum(faktumKey, fakta, null);
+
+		if (faktum && faktum.faktumId && vedlegg && vedlegg.map) {
+			for (const vedleggsForventning of vedlegg) {
+				if (vedleggsForventning.faktumId.toString() === faktum.faktumId.toString()) {
+					return vedleggsForventning.vedleggId;
+				}
+			}
+		}
+		return -1;
+	}
+
 	handleFileUpload(files: FileList) {
-		const vedlegg: VedleggType[] = [];
-		Array.from(files).map((file: File) => {
-			vedlegg.push({ name: file.name});
-		});
-		this.setState({
-			vedlegg
-		});
-		event.preventDefault();
+		const formData = new FormData();
+		for (let i = 0; i < files.length; i++) {
+			formData.append("file" + i, files[i], files[i].name);
+		}
+		this.props.lastOppVedlegg(this.state.vedleggId, formData);
 	}
 
-	render() {
-		return (
-			<div className="container--noPadding">
-				<p>
-					{this.props.label}
-				</p>
-				<Knapp
-					type="standard"
-					htmlType="submit"
-					spinner={false}
-					disabled={false}
-					onClick={() => {
-						this.refs.leggTilVedleggKnapp.click();
-					}}
-				>+ <FormattedMessage id="opplysninger.vedlegg.knapp.tekst"/></Knapp>
+	componentDidUpdate() {
+		if (this.props.featureToggleBeOmLonnslippVedlegg) {
+			if (this.state.vedleggId === null ) {
+				const vedleggId = this.hentVedleggsId(this.props.faktumKey);
+				if (vedleggId > 0) {
+					this.setState({vedleggId});
+				}
+			}
+		}
+	}
 
-				<input
-					ref="leggTilVedleggKnapp"
-					onChange={(e) => this.handleFileUpload(e.target.files)}
-					type="file"
-					className="visuallyhidden"
-					accept="image/jpeg,image/png,application/pdf"
-					multiple={true}
-				/>
-				<div>
-					{this.state.vedlegg.map((vedlegg: VedleggType, index: number) => {
-						return (
-							<div key={index}>
-								<Lenkeknapp
-									onClick={ () => { alert("todo"); }}
-									label={vedlegg.name}
-								/>
-							</div>
+	componentDidMount() {
+		if (this.props.featureToggleBeOmLonnslippVedlegg) {
+			this.props.hentVedleggsForventning();
+		}
+	}
 
-						);
-					})
-					}
+	render() {
+		const vedleggForventet: boolean = (Number(this.state.vedleggId) > 0);
+		if (!vedleggForventet) {
+			return null;
+		} else {
+			return (
+				<div className="container--noPadding">
+					<p>
+						{this.props.label}
+					</p>
+					<VedleggsListe vedlegg={this.props.filer}
+					/>
+					<Knapp
+						type="standard"
+						htmlType="submit"
+						spinner={false}
+						disabled={false}
+						onClick={() => {
+							this.refs.leggTilVedleggKnapp.click();
+						}}
+					>
+						+ <FormattedMessage id="opplysninger.vedlegg.knapp.tekst"/>
+					</Knapp>
+					<input
+						ref="leggTilVedleggKnapp"
```

Comment:
Ikke bruke refs direkte, men ref={ c => this.variable = c}

Reply:
Fikk det ikke til å fungere.